### ISSUE TITLE
NPE - Unexpected exception in CPS VM thread: CpsFlowExecution

### ISSUE THREAD

**Issue by gtirloni** [jenkinsci/github-autostatus-plugin#9]
Jenkins 2.121.2
Github Autostatus Plugin 2.1
GitHub Branch Source Plugin 2.3.6
Github Custom Notification Context SCM Behaviour 1.1
GitHub Pipeline for Blue Ocean 1.7.1
GitHub plugin 1.29.2
Pipeline: GitHub 2.0
Git plugin 3.9.1

```
Jul 21, 2018 4:08:31 PM org.jenkinsci.plugins.githubautostatus.GithubBuildStatusGraphListener log
INFO: Processing build mygithuborg » myrepo » PR-5 #20
Jul 21, 2018 4:08:31 PM org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService reportProblem
WARNING: Unexpected exception in CPS VM thread: CpsFlowExecution[Owner[mygithuborg/myrepo/PR-5/20:gtirloni-gpii/universal/PR-5 #20]]
java.lang.NullPointerException
	at org.jenkinsci.plugins.githubautostatus.GithubNotificationConfig.setGHRepositoryInfo(GithubNotificationConfig.java:191)
	at org.jenkinsci.plugins.githubautostatus.GithubNotificationConfig.fromRun(GithubNotificationConfig.java:105)
	at org.jenkinsci.plugins.githubautostatus.GithubNotificationConfig.fromRun(GithubNotificationConfig.java:89)
	at org.jenkinsci.plugins.githubautostatus.GithubBuildStatusGraphListener.checkEnableBuildStatus(GithubBuildStatusGraphListener.java:244)
	at org.jenkinsci.plugins.githubautostatus.GithubBuildStatusGraphListener.onNewHead(GithubBuildStatusGraphListener.java:72)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.notifyListeners(CpsFlowExecution.java:1438)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$3.run(CpsThreadGroup.java:417)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$1.run(CpsVmExecutorService.java:35)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by jeffpearce** [comment]
Hi @gtirloni. Thanks for reporting this. It should be pretty easy to fix this. I'm wondering whether the plugin is working in general, or whether this happens for every build? Either way I'll fix it, of course

**Comment by gtirloni** [comment]
@jeffpearce It's happening for every build. Thanks a lot for looking into this!

**Comment by jeffpearce** [comment]
@gtirloni - the specific error points to the code having trouble accessing the credentials for github. Does this config per chance have a user id without a password? That's one thing that could cause this

**Comment by gtirloni** [comment]
Yes, we're using the GitHub Branch Source plugin. We've had to add the GitHub credentials in two different ways to satisfy the branch source and the github integration plugins: once with "username and password" (username is GH ID, password is personal access token) and another with "secret text" (ID is username, text is the personal access token).

So in "Configure Jenkins" under "GitHub", we have it using secret text. And when discovering GH orgs, we had to use the secret text one (otherwise the credential won't even show up).

**Comment by jeffpearce** [comment]
@gtirloni I couldn't exactly repro your setup for some reason, but the problem seems pretty strait forward. Are you able to install a private build of the plugin for verification?

**Comment by gtirloni** [comment]
@jeffpearce yes, certainly. thanks!

**Comment by jeffpearce** [comment]
I uploaded to https://1drv.ms/u/s!AoghZSNMKZIQkP9wXF1tav23w5Yv1w. I also published to the experimental plugin update center, but that'll probably take awhile to show up. LMK if it fixes the problem for you

**Comment by gtirloni** [comment]
@jeffpearce hi Jeff, I tested it with a new build and there were not NPE's anymore and the GitHub status were set as expected. Thanks a lot, much appreciated!

**Comment by jeffpearce** [comment]
@gtirloni thanks again for reporting this and verifying. It's fixed in pending release 3.0
